torch/csrc/jit/pickle.cpp

```diff
@@ -0,0 +1,81 @@
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/pickle.h>
+#include <torch/csrc/jit/pickler.h>
+
+
+namespace torch {
+namespace jit {
+
+void pickle_stream(
+    std::function<void(const char*, size_t)> writer,
+    const IValue& ivalue,
+    std::vector<at::Tensor>* tensor_table) {
+  Pickler pickler(std::move(writer), tensor_table);
+
+  if (tensor_table == nullptr) {
+    // No tensor table provided, so tensors will be stored directly in the blob.
+    // Add torch.save metadata so these tensors can be de-serialized later
+    pickler.torchSaveStart();
+  }
+
+  pickler.protocol();
+  pickler.pushIValue(ivalue);
+  pickler.stop();
+
+  if (tensor_table == nullptr) {
+    // No tensor table provided, so tensors will be stored directly in the blob.
+    // Add torch.save metadata so these tensors can be de-serialized later
+    pickler.torchSaveStop();
+  }
+}
+
+std::vector<char> pickle(
+    const IValue& ivalue,
+    std::vector<at::Tensor>* tensor_table) {
+  std::vector<char> data;
+
+  pickle_stream(
+      [&](const char* bytes, size_t len) {
+        data.insert(data.end(), bytes, bytes + len);
+      },
+      ivalue,
+      tensor_table);
+
+  return data;
+}
+
+std::vector<IValue> unpickle(
```

Comment:
This should return a single IValue to be consistent with Pickle. The only reason the current pickler returns a list is because the attributes were in a list, but this is no longer going to be true anyway, so we should refactor the pickler to just produce a single object.